### DISCUSSION TITLE
METviewer Docker image files

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#1146] [Plot Generation] [answered]
I would like to install the latest METViewer version on a DoD HPC and need to access the site (repo) from which I can pull the image file for conversion to a Singularity container. For MET, I have successfully pulled the image file from docker://dtcenter/met:10.0.0, so I assume that using the correct name and version number for METViewer, I could execute the pull similarly. Please advise. Thanks.

**Comment by georgemccabe** [comment, marked as answer]
Hi @jwraby, there are METviewer images available on DockerHub.

https://hub.docker.com/repository/docker/dtcenter/metviewer/tags?page=1&ordering=last_updated

I see that there is a 4.0.0 tag, so docker://dtcenter/metviewer:4.0.0 should correspond to met:10.0.0.

However, I have only set up METviewer in Docker by using docker-compose which also sets up up a mysql container that is connected to the metviewer container. Here is some information about that process: https://github.com/dtcenter/METviewer/blob/main_v4.0/docker/README

I believe the current "latest" version is 4.0.0, so following instructions in the above link should obtain that version. If you need to specify a specific version, I believe that can be done in the METVIEWER_DOCKER_IMAGE environment variable value.

@TatianaBurek, do you have any additional comments?

**Reply by jwraby** [reply]
Hi George, using the info you provided, I successfully pulled/converted the Docker container image file for metviewer 4.0.0 to my account on the HPC. Your assistance is greatly appreciated!

On running the Singularity container I had the following error:
true: /opt/tomcat/bin/startup.sh: Permission denied

If you have any suggestions on the cause of this error, I would appreciate hearing from you. I will create an incident to obtain assistance from the HPC Help Desk to get their help with a resolution since this seems like an issue caused by the environment on the HPC.

**Reply by JohnHalleyGotway** [reply]
John, I suspect that the METviewer container requires root access INSIDE the container. For example, starting Tomcat likely requires privileged access. However with Singularity, the privileges you have OUTSIDE the container are used INSIDE the container. And that'd explain the runtime error.

In fact, we have 2 issues defined on a related project:

https://github.com/NCAR/container-dtc-nwp/issues/11

https://github.com/NCAR/container-dtc-nwp/issues/23

I haven't tried this myself, but a colleague has and has run into issues.

@TatianaBurek anything else to add?

**Comment by jwraby** [comment]
CLASSIFICATION: UNCLASSIFIED

 

Hi John – 

 

Thanks for this info. When I interact with the HPC Help Desk I’ll mention it. So far I haven’t been contacted by them to start troubleshooting.

 

Have a good weekend!

 

R/

John

 

From: johnhg ***@***.***> 
Sent: Thursday, September 9, 2021 5:13 PM
To: dtcenter/METplus ***@***.***>
Cc: jwraby ***@***.***>; Mention ***@***.***>
Subject: [Non-DoD Source] Re: [dtcenter/METplus] METviewer Docker image files (Discussion #1146)

 

All active links contained in this email were disabled. Please verify the identity of the sender, and confirm the authenticity of all links contained within the message prior to copying and pasting the address to a Web browser. 

  _____  

 

John, I suspect that the METviewer container requires root access INSIDE the container.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub < Caution-https://github.com/dtcenter/METplus/discussions/1146#discussioncomment-1304113 > , or unsubscribe < Caution-https://github.com/notifications/unsubscribe-auth/AUX5FH63NHCBTWUMOFJ4IZ3UBE5OJANCNFSM5DS6WEFA > .
Triage notifications on the go with GitHub Mobile for iOS < Caution-https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675 >  or Android < Caution-https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub > .  <Caution-https://github.com/notifications/beacon/AUX5FHZKPF4ESBKZBANFY43UBE5OJA5CNFSM5DS6WEFKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAJ6MMI.gif> 

CLASSIFICATION: UNCLASSIFIED
